### PR TITLE
Support 5.1.11

### PR DESCRIPTION
**Pull request by rcorre**
This adds fully support for the 5.1.11 API. It is not all tested, but I ported over some examples to validate some of the new additions (there's also a Makefile for building the examples).

The `ex_dir` example currently segfaults and I haven't been able to figure out why yet, but I wanted to get this PR up.

### CODE REVIEW

File: allegro5/fshook.d [hunk 1]
```diff
@@ -44,6 +44,9 @@ else
 	import core.stdc.time : time_t;
 }
 
+// keep the callback outside the extern (C) block so it can be a D function
```
Comment (SiegeLord) [inline allegro5/fshook.d]: I forget, does D have the C calling convention? If not, then this won't work.

**Comment by SiegeLord** [comment]
Fixing the  comment above makes ex_dir not segfault for me anymore. One additional thing I noticed is that `al_get_fs_entry_name` has the wrong type. Once you fix that, changing the `print_entry` in ex_dir.d to call `fromStringz` on the returned name should make everything print correctly.

**Comment by rcorre** [comment]
@SiegeLord yup, you're right. The callback needs to be `extern(C)`.

**Comment by SiegeLord** [comment]
Great, thanks!

**Pull request merged**